clinica/pipelines/deeplearning_prepare_data/deeplearning_prepare_data_utils.py

```diff
@@ -0,0 +1,218 @@
+# coding: utf8
+
+def extract_slices(preprocessed_T1, slice_direction=0, slice_mode='original'):
+    """This function extracts the slices from three directions
+
+    Args:
+        preprocessed_T1: tensor version of the nifty MRI.
+        slice_direction: which axis direction that the slices were extracted
+        slice_mode: 'original' or 'RGB'
+
+    Returns:
+        file: tensor saved on the disk.
+    """
+    import torch
+    import os
+
+    image_tensor = torch.load(preprocessed_T1)
+    # reshape the tensor, delete the first dimension for slice-level
+    image_tensor = image_tensor.view(image_tensor.shape[1], image_tensor.shape[2], image_tensor.shape[3])
+
+    # sagital
+    # M and N correspond to the first and last slices (if need to remove)
+    M = 0
+    N = 0
+    slice_list_sag = range(M, image_tensor.shape[0] - N)  # delete the first M slices and last N slices
+
+    basedir = os.getcwd()
+    output_file_original = []
+    output_file_rgb = []
+    if slice_direction == 0:
+        for index_slice, index_slice_list in zip(slice_list_sag, range(len(slice_list_sag))):
+            # for i in slice_list:
+            # sagital
+            slice_select_sag = image_tensor[index_slice, :, :]
+
+            # convert the slices to images based on if transfer learning or not
+            # train from scratch
+            extracted_slice_original_sag = slice_select_sag.unsqueeze(0)  # shape should be 1 * W * L
+
+            # train for transfer learning, creating the fake RGB image.
+            slice_select_sag = (slice_select_sag - slice_select_sag.min()) / (slice_select_sag.max() - slice_select_sag.min())
+            extracted_slice_rgb_sag = torch.stack((slice_select_sag, slice_select_sag, slice_select_sag))  # shape should be 3 * W * L
+
+            # save into .pt format
+            if slice_mode == 'original':
+                output_file_original.append(
+                        os.path.join(
+                            basedir,
+                            os.path.basename(preprocessed_T1).split('.pt')[0]
+                            + '_axis-sag_originalslice-'
+                            + str(index_slice)
```

Comment:
For slices & patches, I was wondering if we should use the same number of characters e.g. `slice-001`, `slice-010`, `slice-100` instead of  `slice-1`, `slice-10`, `slice-100`. What do you think @mdiazmel?

Reply:
I don't think that this would be a improvement. If we change it, how many characters we may propose: 2, 3 (maybe someday 4)...? I prefer to keep it like that.